File: DSA/VaniAgrawal_590015846/Day_90_Question_2.py
# Your task: Implement degree-based analysis to determine existence of Eulerian Path or Circuit using vertex degree counting and mathematical conditions.
def eulerian_path_circuit():
    num_vertices = int(input("Enter the number of vertices: "))
    edges_str = input("Enter the edges as a list of lists, e.g., [[0,1],[1,2],[2,0],[0,3]]: ")
    edges = eval(edges_str)
    
    degrees = [0] * num_vertices
    for u, v in edges:
        degrees[u] += 1
        degrees[v] += 1
    
    odd_degree_count = 0
    for degree in degrees:
        if degree % 2 != 0:
            odd_degree_count += 1
            
    if odd_degree_count == 0:
        print("Output: Eulerian Circuit Exists")
    elif odd_degree_count == 2:
        print("Output: Eulerian Path Exists")
    else:
        print("Output: No Eulerian Path or Circuit Exists")

eulerian_path_circuit()

File: DSA/RATNA_KHANNA_590013141/DAY_90_QUESTION_2.c
#include <stdio.h>

int graph[100][100], degree[100], n;

int main() {
    int m, u, v, odd = 0;
    printf("Enter number of vertices: ");
    scanf("%d", &n);
    printf("Enter number of edges: ");
    scanf("%d", &m);
    for (int i = 0; i < n; i++) {
        degree[i] = 0;
        for (int j = 0; j < n; j++)
            graph[i][j] = 0;
    }
    printf("Enter edges (u v):\n");
    for (int i = 0; i < m; i++) {
        scanf("%d %d", &u, &v);
        graph[u][v] = 1;
        graph[v][u] = 1;
        degree[u]++;
        degree[v]++;
    }
    for (int i = 0; i < n; i++)
        if (degree[i] % 2 != 0)
            odd++;
    if (odd == 0)
        printf("Eulerian Circuit Exists\n");
    else if (odd == 2)
        printf("Eulerian Path Exists\n");
    else
        printf("No Eulerian Path or Circuit\n");
    return 0;
}
